backend/src/main/kotlin/blue/mild/covid/vaxx/service/IsinRetryService.kt

```diff
@@ -0,0 +1,153 @@
+package blue.mild.covid.vaxx.service
+
+import blue.mild.covid.vaxx.dao.repository.PatientRepository
+import blue.mild.covid.vaxx.dto.internal.PatientValidationResult
+import blue.mild.covid.vaxx.dto.request.IsinJobDtoIn
+import blue.mild.covid.vaxx.dto.response.IsinJobDtoOut
+import blue.mild.covid.vaxx.dto.response.PatientDtoOut
+import blue.mild.covid.vaxx.dto.response.toPatientVaccinationDetailDto
+import mu.KLogging
+
+
+class IsinRetryService(
+    private val patientService: PatientService,
+    private val patientValidation: PatientValidationService,
+    private val isinService: IsinServiceInterface,
+    private val dataCorrectnessService: DataCorrectnessService,
+    private val patientRepository: PatientRepository,
+    private val vaccinationService: VaccinationService,
+) {
+
+    private companion object : KLogging()
+
+    @Suppress("ComplexCondition", "ComplexMethod") // this is complex job, it is ok here
+    suspend fun runIsinRetry(isinJobDto: IsinJobDtoIn): IsinJobDtoOut {
+        val stats = IsinJobDtoOut()
+
+        val patients = patientService.getPatientsByConjunctionOf(
+            n = isinJobDto.patientsCount,
+            offset = isinJobDto.patientsOffset.toLong()
+        )
+
+        for (patient in patients) {
```

Comment:
co kdyz tohle nekde vyhodi exception? spis bych kazdeho pacienta validoval zvlast pod `runCatching` aby kdyz jeden bude vytrvale padat, abychom alespon zbytek zkusili do toho ISINu prorvat

Reply:
Kvůli jakým exceptions by to bylo? Http exceptions se tam už handlují, takže by to nikdy nemělo selhat.

Reply:
nevim, v tech retry fcich mas nejake require not null, pak tam mas  personalNumber = patient.personalNumber ?: throw AssertionError { "Personal number cannot be null."},

Reply:
no buh vi co se tam muze stat - snazim se byt co nejvice defensivni, tedy ackoliv neni neco, co by se melo pokonit, je lepsi si porad rikat, ze se to muze stat

Reply:
Jo, ale k těm require not null a assertion error se to nemůže dostat, protože to v té service handluji podmínkami. Pokud to selže, znamená to, že je tam nějaký bug a k tomu přeci ty exceptions jsou, aby to nareportovali? Navíc tohle api budeme spouštět my, tedy když už by to náhodou selhalo, není problém bug opravit, ne?

Reply:
jo, to je pravda, budeme to spoustet my, tak asi ok